iidm/iidm-criteria/src/main/java/com/powsybl/iidm/criteria/translation/DefaultNetworkElementAdapter.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.criteria.translation;
+
+import com.powsybl.iidm.criteria.NetworkElementCriterion.NetworkElementCriterionType;
+import com.powsybl.iidm.network.*;
+
+import java.util.Optional;
+
+/**
+ * <p>Adapter to have a {@link NetworkElement} from an {@link Identifiable} object.</p>
+ * @author Sophie Frasnedo {@literal <sophie.frasnedo at rte-france.com>}
+ */
+public class DefaultNetworkElementAdapter implements NetworkElement {
+
+    private final Identifiable<?> identifiable;
+
+    public DefaultNetworkElementAdapter(Identifiable<?> identifiable) {
+        this.identifiable = identifiable;
+    }
+
+    @Override
+    public String getId() {
+        return identifiable.getId();
+    }
+
+    @Override
+    public Country getCountry1() {
+        return getCountry(TwoSides.ONE);
+    }
+
+    @Override
+    public Country getCountry2() {
+        return getCountry(TwoSides.TWO);
+    }
+
+    @Override
+    public Country getCountry() {
+        switch (identifiable.getType()) {
+            case TWO_WINDINGS_TRANSFORMER -> {
+                Optional<Substation> substation = ((TwoWindingsTransformer) identifiable).getSubstation();
+                return substation.map(Substation::getNullableCountry).orElse(null);
+            }
+            case THREE_WINDINGS_TRANSFORMER -> {
+                Optional<Substation> substation = ((ThreeWindingsTransformer) identifiable).getSubstation();
+                return substation.map(Substation::getNullableCountry).orElse(null);
+            }
+            default -> {
+                Country country1 = getCountry1();
+                return country1 != null ? country1 : getCountry2();
+            }
+
+        }
+    }
+
+    private Country getCountry(TwoSides side) {
+        return switch (identifiable.getType()) {
+            case LINE -> getCountryFromTerminal(((Line) identifiable).getTerminal(side));
+            case TIE_LINE -> getCountryFromTerminal(((TieLine) identifiable).getTerminal(side));
+            default -> null;
+        };
+    }
+
+    private Country getCountryFromTerminal(Terminal terminal) {
+        Optional<Substation> substation = terminal.getVoltageLevel().getSubstation();
+        return substation.map(Substation::getNullableCountry).orElse(null);
+    }
+
+    @Override
+    public Double getNominalVoltage1() {
+        return getNominalVoltage(ThreeSides.ONE);
+    }
+
+    @Override
+    public Double getNominalVoltage2() {
```

Comment:
Maybe it should be Optional ?

Reply:
The `getNominalVoltage…` and `getCountry…` methods now return `Optional`s.